diff --git a/lib/reline/terminfo.rb b/lib/reline/terminfo.rb
index f53642b..78b5476 100644
--- a/lib/reline/terminfo.rb
+++ b/lib/reline/terminfo.rb
@@ -15,17 +15,7 @@ module Reline::Terminfo
   class TerminfoError < StandardError; end
 
   def self.curses_dl_files
-    case RUBY_PLATFORM
-    when /mingw/, /mswin/
-      # aren't supported
-      []
-    when /cygwin/
-      %w[cygncursesw-10.dll cygncurses-10.dll]
-    when /darwin/
-      %w[libncursesw.dylib libcursesw.dylib libncurses.dylib libcurses.dylib]
-    else
-      %w[libncursesw.so libcursesw.so libncurses.so libcurses.so]
-    end
+    %w[@curses_lib@]
   end
 
   @curses_dl = false
